pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSink.java

```diff
@@ -49,7 +54,11 @@
     @Override
     public void write(Record<byte[]> sourceRecord) {
         KeyValue<K, V> keyValue = extractKeyValue(sourceRecord);
-        ProducerRecord<K, V> record = new ProducerRecord<>(kafkaSinkConfig.getTopic(), keyValue.getKey(), keyValue.getValue());
+        Pair<Schema, Schema> keyValueSchemas = extractKeyValueSchemas(sourceRecord);
```

Comment:
I don't think this change is related to the adopter change. Can you explain why do you put the change here?